keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/SecurityGroupHandler.kt

```diff
@@ -219,17 +193,49 @@ open class SecurityGroupHandler(
       }
     }
 
-  override suspend fun actuationInProgress(resource: Resource<SecurityGroupSpec>): Boolean =
-    resource
-      .spec
-      .locations
-      .regions
-      .map { it.name }
-      .any { region ->
-        orcaService
-          .getCorrelatedExecutions("${resource.id}:$region")
-          .isNotEmpty()
+  /**
+   * Returns a map of region names to the corresponding [SecurityGroup] objects to be used in [export].
+   */
+  protected open suspend fun getSecurityGroupsByRegion(exportable: Exportable): Map<String, SecurityGroup> {
```

Comment:
This is the new method, basically just extracting this bit of the logic out of `export` into a separate method that can be overridden by the sub-class.